frontend/packages/operator-lifecycle-manager/src/components/subscription.tsx

```diff
@@ -451,7 +451,7 @@ export class SubscriptionUpdates extends React.Component<
             <dl className="co-m-pane__details">
               <dt className="co-detail-table__section-header">Channel</dt>
               <dd>
-                {this.state.waitingForUpdate ? (
+                {this.state.waitingForUpdate || !pkg ? (
```

Comment:
Do we need to differentiate between loading and load error or not found? Otherwise the loading icon could be present forever. We might just disable the button since we have a value to show from `obj`, but I'm not sure how Patternfly handles disabled buttons of type `link`.

I'd think we'd want a similar for install plan approvals below.

Reply:
I did the change. It however still does not address the issue. It does not show error as mentioned in the bug by disabling the button but the problem I see is that the loading of PackageManifest still takes ages (payload is around 1.6MB in postman). Now the user doesn't even have a loading icon for him to accurately reflect the state(arbitrarily wait for the link to be active).  
It only solves the error but the user still has to wait or refresh the page for that button to come alive. 
One alternative that I tried which improved the speed was narrowing the query to a particular manifest(using the name of the operator). But then again the solution is not an elegant one.

Reply:
> I did the change. It however still does not address the issue. It does not show error as mentioned in the bug by disabling the button but the problem I see is that the loading of PackageManifest still takes ages (payload is around 1.6MB in postman). Now the user doesn't even have a loading icon for him to accurately reflect the state(arbitrarily wait for the link to be active).

You change channels rarely. With the loading icon, it means you can't even *see* the value until the package manifest request completes. So we've made the common case (just looking at the details) much slower.

The size of the response is the real problem. We might want to look at gzipping API responses through the console proxy.

> One alternative that I tried which improved the speed was narrowing the query to a particular manifest(using the name of the operator). But then again the solution is not an elegant one.

This is a much better solution if there is a reliable field selector or label selector we can use.

Reply:
Obviously not for this PR as it's a significant architectural change, but GraphQL could really help here long term.

Reply:
The only reliable selector is the name of the operator however since the name is in the `name-version-release` we will have to perform string manipulation on it to get the `name` part which the package manifest uses. Your thoughts?

Reply:
@spadgett ^^